doc/designs/expiring-password-notification.md

```diff
@@ -0,0 +1,217 @@
+# Expiring Password Notifications
+
+**DESIGN STAGE**
+
+## Overview
+
+A method to warn users via email that their IPA account password is about to expire.
+Ticket [link](https://pagure.io/freeipa/issue/3687).
+
+## User Stories
+
+[0] As an IPA user, I want to be notified by email and through the WebUI when my password is near its expiry date so that I change my password before it expires.
+
+Outcome: Users whose passwords are expiring receive an email.
+NB: The WebUI warning already exists. Document (ipapwdexpadvnotify) usage.
```

Comment:
https://fedorahosted.org/freeipa/ticket/2625

Reply:
Added for completeness (it was already in the documentation section), thanks!